### PR TITLE
Feature/middeware - CheckRecipient

### PR DESCRIPTION
**Pull request by dankelleher**
Adds a simple middleware: "CheckRecipient".

This middleware enforces that transactions from the cryptid account must be SystemProgram.Transfer only, and must be sent to the designated recipient.

This PR also introduces the "obligatory middleware" model or a cryptid PDA that is derived from a middleware account, such that the cryptid account enforces use of the middleware, and the middleware cannot be removed.

With this PR, obligatory middleware is the only model supported. Also note - middleware is not supported by direct_execute.

### CODE REVIEW

File: cryptid_anchor/programs/middleware/check_recipient/src/lib.rs [hunk 1]
```diff
@@ -0,0 +1,142 @@
+extern crate core;
+
+use anchor_lang::prelude::*;
+use cryptid_anchor::cpi::accounts::ApproveExecution;
+use cryptid_anchor::program::CryptidAnchor;
+use cryptid_anchor::state::transaction_account::TransactionAccount;
+
+declare_id!("midcHDoZsxvMmNtUr8howe8MWFrJeHHPbAyJF1nHvyf");
```
Comment (rado0x54) [inline cryptid_anchor/programs/middleware/check_recipient/src/lib.rs:8]: you grinded for `midc` here?

File: cryptid_anchor/programs/middleware/check_recipient/src/lib.rs [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+        bump,
+    )]
+    pub middleware_account: Account<'info, CheckRecipient>,
+    #[account(mut)]
+    pub authority: Signer<'info>,
+    pub system_program: Program<'info, System>,
+}
+
+#[derive(Accounts)]
+pub struct ExecuteMiddleware<'info> {
```
Comment (rado0x54) [inline cryptid_anchor/programs/middleware/check_recipient/src/lib.rs:81]: I will wait for the architecture meeting, but is the chaining element already in this version? No, correct?

File: cryptid_anchor/programs/middleware/check_recipient/src/lib.rs (hunk 1 above)
Comment (dankelleher) [reply cryptid_anchor/programs/middleware/check_recipient/src/lib.rs:8]: yes - we can regrind for something nicer if we like. I just wanted something temporarily easy to recognise.

File: cryptid_anchor/programs/middleware/check_recipient/src/lib.rs (hunk 2 above)
Comment (dankelleher) [reply cryptid_anchor/programs/middleware/check_recipient/src/lib.rs:81]: No, no chaining yet

**Pull request merged**